x-pack/plugin/sql/jdbc/src/main/java/org/elasticsearch/xpack/sql/jdbc/TypeConverter.java

```diff
@@ -210,6 +209,15 @@ static long convertFromCalendarToUTC(long value, Calendar cal) {
         return failConversion(val, columnType, typeString, type);
     }
 
+    static List<Object> convertArray(Object v, EsType columnBaseType, String typeString) throws SQLException {
+        List<Object> values = new ArrayList<>();
+        for (Object o : (List<?>) v) {
```

Comment:
Unchecked cast from object to list - shouldn't this fail?

Reply:
I think that's only a warning, but would be nice to add the `@SuppressWarnings("unchecked")`

Reply:
The cast failure is checked (by [code](https://github.com/bpintea/elasticsearch/blob/edba5cdeea8da15aa8e13ea6632356d335646173/x-pack/plugin/sql/jdbc/src/test/java/org/elasticsearch/xpack/sql/jdbc/TypeConverterTests.java#L106)), but yes, not sure why isn't a warning emitted by the compiler; editor labels the annotation as redundant, if added.